src/formatter/javascript.ts

```diff
@@ -73,7 +80,25 @@ export class SyntheticsGenerator extends JavaScriptLanguageGenerator {
     this.previousContext = undefined;
   }
 
-  generateAction(actionInContext: ActionInContext) {
+  /**
+   * Generate code for an action.
+   * @param actionInContext The action to create code for.
+   * @param ignoreStepDefaults When true, the function will not create steps between
+   * navigation events as it normally would.
+   * @param customStepsOverride When true, the function will not throw an error if the
+   * called while the generator is already recording a step.
+   * @returns the strings generated for the action.
+   */
+  generateAction(
+    actionInContext: ActionInContext,
+    ignoreStepDefaults = false,
```

Comment:
Can we be more specific on the parameter name, for example, `newStepOnNavigate` or something so users don't need to remember the default behaviour? (I appreciate you've provided a great description, but I still prefer if I didn't need to read them :D)